iree/turbine/kernel/wave/wave.py

```diff
@@ -521,23 +519,33 @@ def test_execute(self, args, kwargs):
         config = kwargs.get("run_config", None)
         use_scheduling = kwargs.get("schedule", False)
         use_scheduling_barriers = kwargs.get("use_scheduling_barriers", False)
+        # When this is passed in from the user, we will populate it with the kernel hash.
+        # It will always be returned with just one entry which is the hash of the kernel.
+        cached_kernel_hash = kwargs.get("kernel_hash", [])
 
         # Get cached kernel when available.
         cache_enabled = is_cache_enabled()
+        kernel_hash = None
         if cache_enabled:
-            cache_manager = get_cache_manager()
             # TODO: Move use_scheduling, use_scheduling_barriers, etc. into the config so everything is contained there.
-            processed_constraints = anonymize_constraints(self.constraints)
-            kernel_hash = cache_manager.get_hash(
-                processed_constraints,
-                self._f,
-                safe_dict_to_tuple(IndexingContext.current().subs),
-                tuple(dynamic_symbols),
-                safe_dict_to_tuple(config),
-                use_scheduling=use_scheduling,
-                use_scheduling_barriers=use_scheduling_barriers,
-                run_bench=run_bench,
-            )
+            cache_manager = get_cache_manager()
+            if cached_kernel_hash:
+                # If a cached_kernel_hash is passed in, we assume it was generated in a previous run
+                # and since we always return a list of size 1, we can just grab the first element.
+                kernel_hash = cached_kernel_hash[0]
```

Comment:
just double checking, so it wouldn't be possible for previous `kernel_hash` to != `cached_kernel_hash[0]`?

Reply:
That is possible. Essentially what this new arg does is allow the user to see what the kernel hash is for a given kernel. The user could actually run 5 different kernels and get their hashes and then decide which kernel they want to run by choosing the appropriate hash and feeding it back in. So its not always guaranteed to be the previous one. Since it is passed in from the user, it could also contain more than one hash. But we only use the first one because that is consistent with what we passed back to the user.